_source/_posts/2022-06-17-simple-crud-react-and-spring-boot.md

```diff
@@ -0,0 +1,1350 @@
+      let updatedGroups = [...groups].filter(i => i.id !== id);
+      setGroups(updatedGroups);
+    });
+  }
+
+  if (loading) {
+    return <p>Loading...</p>;
+  }
+
+  const groupList = groups.map(group => {
+    const address = `${group.address || ''} ${group.city || ''} ${group.stateOrProvince || ''}`;
+    return <tr key={group.id}>
+      <td style={{whiteSpace: 'nowrap'}}>{group.name}</td>
+      <td>{address}</td>
+      <td>{group.events.map(event => {
+        return <div key={event.id}>{new Intl.DateTimeFormat('en-US', {
+          year: 'numeric',
+          month: 'long',
+          day: '2-digit'
+        }).format(new Date(event.date))}: {event.title}</div>
+      })}</td>
+      <td>
+        <ButtonGroup>
+          <Button size="sm" color="primary" tag={Link} to={"/groups/" + group.id}>Edit</Button>
+          <Button size="sm" color="danger" onClick={() => remove(group.id)}>Delete</Button>
+        </ButtonGroup>
+      </td>
+    </tr>
+  });
+
+  return (
+    <div>
+      <AppNavbar/>
+      <Container fluid>
+        <div className="float-end">
+          <Button color="success" tag={Link} to="/groups/new">Add Group</Button>
+        </div>
+        <h3>My JUG Tour</h3>
+        <Table className="mt-4">
+          <thead>
+          <tr>
+            <th width="20%">Name</th>
+            <th width="20%">Location</th>
+            <th>Events</th>
+            <th width="10%">Actions</th>
+          </tr>
+          </thead>
+          <tbody>
+          {groupList}
+          </tbody>
+        </Table>
+      </Container>
+    </div>
+  );
+};
+
+export default GroupList;
+```
+{% endraw %}
+
+Create `AppNavbar.js` in the same directory to establish a common UI feature between components.
+
+{% raw %}
+```jsx
+import React, { useState } from 'react';
+import { Collapse, Nav, Navbar, NavbarBrand, NavbarToggler, NavItem, NavLink } from 'reactstrap';
+import { Link } from 'react-router-dom';
+
+const AppNavbar = () => {
+
+  const [isOpen, setIsOpen] = useState(false);
+
+  return (
+    <Navbar color="dark" dark expand="md">
+      <NavbarBrand tag={Link} to="/">Home</NavbarBrand>
+      <NavbarToggler onClick={() => { setIsOpen(!isOpen) }}/>
+      <Collapse isOpen={isOpen} navbar>
+        <Nav className="justify-content-end" style={{width: "100%"}} navbar>
+          <NavItem>
+            <NavLink href="https://twitter.com/oktadev">@oktadev</NavLink>
+          </NavItem>
+          <NavItem>
+            <NavLink href="https://github.com/oktadev/okta-spring-boot-react-crud-example">GitHub</NavLink>
+          </NavItem>
+        </Nav>
+      </Collapse>
+    </Navbar>
+  );
+};
+
+export default AppNavbar;
+```
+{% endraw %}
+
+Create `app/src/Home.js` to serve as the landing page for your app.
+
+```jsx
+import React from 'react';
+import './App.css';
+import AppNavbar from './AppNavbar';
+import { Link } from 'react-router-dom';
+import { Button, Container } from 'reactstrap';
+
+const Home = () => {
+  return (
```

Comment:
can be implicit return

Reply:
I'm not sure what you mean by this. Can you show me with a suggested change?

Reply:
I figured it out. Currently, the code is:

```jsx
const Home = () => {
  return (
    <div>
      <AppNavbar/>
      <Container fluid>
        <Button color="link"><Link to="/groups">Manage JUG Tour</Link></Button>
      </Container>
    </div>
  );
}
```

And it could be changed to:

```jsx
const Home = () =>
  (
    <div>
      <AppNavbar/>
      <Container fluid>
        <Button color="link"><Link to="/groups">Manage JUG Tour</Link></Button>
      </Container>
    </div>
  )
```

As a Java developer, the 2nd makes more sense to me, so I think I'll leave it. One neat thing I discovered is IntelliJ can refactor from the first to the 2nd for you. 

<img width="861" alt="Screen Shot 2022-06-17 at 08 52 50" src="https://user-images.githubusercontent.com/17892/174322828-bc58c618-284f-4f37-b482-f06a18176943.png">